gems/aws-sdk-core/lib/aws-sdk-core/rpc_v2/error_handler.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Aws
+  module RpcV2
+    class ErrorHandler < Aws::ErrorHandler
+
+      def call(context)
+        @handler.call(context).on(300..599) do |response|
+          response.error = error(context)
+          response.data = nil
+        end
+      end
+
+      private
+
+      def extract_error(body, context)
+        body = body.force_encoding(Encoding::BINARY)
```

Comment:
What happens if we don't do this?

Reply:
Eventually we get a `ArgumentError: invalid byte sequence in UTF-8`

Not sure where else this goes.

We do it in parsing data too:

```
        cbor_data = context.http_response.body_contents.force_encoding(Encoding::BINARY)

```